grype/pkg/syft_sbom_provider.go

```diff
@@ -23,67 +34,126 @@ func (e errEmptySBOM) Error() string {
 }
 
 func syftSBOMProvider(userInput string, config ProviderConfig) ([]Package, Context, error) {
-	reader, err := getSBOMReader(userInput)
+	s, err := getSBOM(userInput, config)
 	if err != nil {
 		return nil, Context{}, err
 	}
 
-	sbom, format, err := syft.Decode(reader)
+	return FromCatalog(s.Artifacts.PackageCatalog, config), Context{
+		Source: &s.Source,
+		Distro: s.Artifacts.LinuxDistribution,
+	}, nil
+}
+
+func newInputInfo(scheme, contentTye string) *inputInfo {
+	return &inputInfo{
+		Scheme:      scheme,
+		ContentType: contentTye,
+	}
+}
+
+type inputInfo struct {
+	ContentType string
+	Scheme      string
+}
+
+func getSBOM(userInput string, config ProviderConfig) (*sbom.SBOM, error) {
+	reader, info, err := getSBOMReader(userInput, config)
```

Comment:
I think we don't want to change the signature here. `getSBOMReader` should not need to return the extra info here . `getSBOMReader` can be given to the agency to determine which error it returns based on the input discovered.